mapo/envs/navigation.py

```diff
@@ -110,11 +110,11 @@ def _transition_log_prob_fn(self, state, action, next_state):
     def _reward_fn(self, state, action, next_state):
         # pylint: disable=invalid-unary-operand-type
         goal = tf.constant(self._end, name="goal")
-        return -tf.norm(state - goal, axis=-1)
+        return -tf.norm(next_state - goal, axis=-1)
```

Comment:
Need to condition on the next state in order to reach the goal in 20 steps. Furthermore, by conditioning on the current state we would always receive the same penalty in the first time step, regardless of the agent's decision.

Reply:
This is a bit controversial from the perspective of the Bellman equation, but I don't see any problem about using the `next_state` here.